app/javascript/controllers/opening_time_controller.js

```diff
@@ -0,0 +1,82 @@
+import { Controller } from "stimulus";
+
+export default class extends Controller {
+  static targets = [
+    "opensAt",
+    "closesAt",
+    "timesBlock",
+    "closed",
+    "closedBlock",
+    "open24h",
+    "open24hBlock"
+  ]
+
+  initialize() {
+    this.toggleClosed = this.toggleClosed.bind(this)
+    this.toggleOpen24h = this.toggleOpen24h.bind(this)
```

Comment:
Why do you need to bind it?

Reply:
To be able to refer to the Stimulus controller in the `toggleClosed` / `toggleOpen24h` methods. Without it, we can't call `this.timesBlockTarget` etc.